Sources_v3/StreamChatUI/ChatChannelList/ChatUnreadCountBubbleView.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class UnreadCountBubbleView: UIView {
+    
+    let inset: CGFloat = 4
+    
+    open override var intrinsicContentSize: CGSize {
+        .init(width: max(unreadCountLabel.intrinsicContentSize.width + inset*2, 16), height: 16)
+    }
+    
+    var unreadCount: ChannelUnreadCount = .noUnread {
+        didSet {
+            updateContent()
+        }
+    }
+    
+    private let unreadCountLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    // MARK: - Init
+    
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+    
+    private func commonInit() {
+        setupLayout()
+        setupAppearance()
+        updateContent()
+    }
+    
+    // MARK: - Layout
+    
+    override open func invalidateIntrinsicContentSize() {
+        super.invalidateIntrinsicContentSize()
+        
+        unreadCountLabel.invalidateIntrinsicContentSize()
+    }
+    
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+        
+        invalidateIntrinsicContentSize()
+        
+        layer.cornerRadius = intrinsicContentSize.height / 2
+    }
+    
+    // MARK: - Public
+    
+    open func setupAppearance() {
+        layer.masksToBounds = true
+        backgroundColor = .init(r: 255, g: 55, b: 66)
+        unreadCountLabel.backgroundColor = .clear
+        unreadCountLabel.textColor = .white
+        unreadCountLabel.font = .boldSystemFont(ofSize: .scaledFontSize(for: 11))
+        unreadCountLabel.textAlignment = .center
+        
+    }
+    
+    open func setupLayout() {
+        embed(unreadCountLabel, insets: .init(top: inset, leading: inset, bottom: -inset, trailing: -inset))
+    }
+    
+    open func updateContent() {
+        if isMock {
+            unreadCountLabel.text = ["4", "42", "420"].randomElement()
+        } else {
```

Comment:
Can we somehow incapsulate `isMock == true` check in a controller and provide a mock/updated channel? Just for the sake of having the mock stuff in a single place.

Reply:
I planned this `isMock` to be only `WIP` solution 😄